src/app/Exec/lib/backend_proxy.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+const { BackendProxy: BackendProxyBase } = require("backend");
+const DirectBackend = require("./backends/direct/index");
+const JenkinsBackend = require("./backends/jenkins/index");
+const Backend = require("./types/backend");
+
+const BackendTypes = {
+    direct: DirectBackend,
+    jenkins: JenkinsBackend
+};
+
+class BackendProxy extends BackendProxyBase {
+    constructor() {
+        super(Backend);
+    }
+
+    createExecutor(backend, data = null) {
```

Comment:
I think that this method should set backendId/backend on created executor!
Set data to empty object by default. And further down below, do
```js
const createData = Object.assign({ backendId: backend }, data);
return new this.executorClasses[backendType](createData);
```